torch/_dynamo/variables/builder.py

```diff
@@ -671,6 +673,17 @@ def index_source(key):
             return self.tx.output.side_effects.track_object_existing(
                 self.source, value, result
             )
+        elif isinstance(value, types.GetSetDescriptorType):
+            return GetSetDescriptorVariable(
+                value, guards=self.make_guards(GuardBuilder.ID_MATCH)
```

Comment:
I think this is better as a FUNCTION_MATCH. As written, we'll keep repeatedly checking if torch.Tensor.dtype has been monkeypatched which is inconsistent with our current global function policy

Reply:
Good catch, I didn't realize we didn't check global functions